datatypes/src/raster/geo_transform.rs

```diff
@@ -380,6 +414,40 @@ mod tests {
         );
     }
 
+    #[test]
+    fn lower_right_pixel_index_edge() {
+        let geo_transform = GeoTransform::test_default();
+
+        let partition = SpatialPartition2D::new((1., 1.).into(), (8., -8.).into()).unwrap();
+        assert_eq!(
+            geo_transform.lower_right_pixel_idx(&partition),
+            [7, 7].into()
+        );
+
+        let partition = SpatialPartition2D::new((1., 1.).into(), (8.5, -8.).into()).unwrap();
+        assert_eq!(
+            geo_transform.lower_right_pixel_idx(&partition),
+            [7, 8].into()
+        );
+
+        let partition = SpatialPartition2D::new((1., 1.).into(), (8., -8.5).into()).unwrap();
+        assert_eq!(
+            geo_transform.lower_right_pixel_idx(&partition),
+            [8, 7].into()
+        );
+    }
+
+    #[test]
+    fn lower_right_pixel_index_inside() {
+        let geo_transform = GeoTransform::test_default();
+
+        let partition = SpatialPartition2D::new((1., 1.).into(), (7.5, -7.5).into()).unwrap();
```

Comment:
We need some much smaller numbers wrt. the epsilon.

Reply:
the epsilon is a factor that is multiplied with the pixel_size. This is not new. I moved it from tiling_strategy